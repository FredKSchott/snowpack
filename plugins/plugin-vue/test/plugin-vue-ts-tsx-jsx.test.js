const path = require('path');
const plugin = require('../plugin');

test('plugin vue with ts', async () => {
  const pluginInstance = plugin({
    buildOptions: {
      sourceMap: true,
    },
  });
  const pluginLoad = pluginInstance.load;
  const codeFilePath = path.resolve(__dirname, './stubs/VueContentTs.vue');
  const resultContent = await pluginLoad({
    filePath: codeFilePath,
  });
  expect(resultContent).toMatchSnapshot();
});

test('plugin vue with tsx', async () => {
  const pluginInstance = plugin({
    buildOptions: {
      sourceMap: true,
    },
  });
  const pluginLoad = pluginInstance.load;
  const codeFilePath = path.resolve(__dirname, './stubs/VueContentTsx.vue');
  const resultContent = await pluginLoad({
    filePath: codeFilePath,
  });
  expect(resultContent).toMatchSnapshot();
});

test('plugin vue with jsx', async () => {
  const pluginInstance = plugin({
    buildOptions: {
      sourceMap: true,
    },
  });
  const pluginLoad = pluginInstance.load;
  const codeFilePath = path.resolve(__dirname, './stubs/VueContentJsx.vue');
  const resultContent = await pluginLoad({
    filePath: codeFilePath,
  });
  expect(resultContent).toMatchSnapshot();
});
