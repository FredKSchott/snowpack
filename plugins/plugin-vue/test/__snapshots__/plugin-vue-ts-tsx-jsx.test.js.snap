// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`plugin vue with jsx 1`] = `
Object {
  ".js": Object {
    "code": "import { Fragment } from 'vue';
import {createVNode, isVNode} from 'vue';
const slice = Array.prototype.slice;
export function jsx(tag, props = null, children = null) {
  if (arguments.length > 3 || isVNode(children)) {
    children = slice.call(arguments, 2);
  }
  return createVNode(tag, props, children);
}

import {defineComponent} from \\"vue\\";
export default defineComponent({
  name: \\"VueContentJsx\\",
  setup() {
    return () => /* @__PURE__ */ jsx(Fragment, null, /* @__PURE__ */ jsx(\\"h1\\", null, \\"VueContentJsx\\"));
  }
});",
    "map": "",
  },
}
`;

exports[`plugin vue with ts 1`] = `
Object {
  ".js": Object {
    "code": "import {defineComponent} from \\"vue\\";
const defaultExport = defineComponent({
  name: \\"VueContentTs\\",
  setup() {
    const name = \\"VueContentTs\\";
  }
});
import { createVNode as _createVNode, openBlock as _openBlock, createBlock as _createBlock } from \\"vue\\"

export function render(_ctx, _cache) {
  return (_openBlock(), _createBlock(\\"h1\\", null, \\"Vue Content Ts\\"))
}

defaultExport.render = render
export default defaultExport",
    "map": "",
  },
}
`;

exports[`plugin vue with tsx 1`] = `
Object {
  ".js": Object {
    "code": "import { Fragment } from 'vue';
import {createVNode, isVNode} from 'vue';
const slice = Array.prototype.slice;
export function jsx(tag, props = null, children = null) {
  if (arguments.length > 3 || isVNode(children)) {
    children = slice.call(arguments, 2);
  }
  return createVNode(tag, props, children);
}

import {defineComponent} from \\"vue\\";
export default defineComponent({
  name: \\"VueContentTsx\\",
  setup() {
    const name = \\"VueContentTsx\\";
    return () => /* @__PURE__ */ jsx(Fragment, null, /* @__PURE__ */ jsx(\\"h1\\", null, \\"VueContentTsx\\"));
  }
});",
    "map": "",
  },
}
`;
